pyiron/sphinx/base.py

```diff
@@ -1515,6 +1522,93 @@ def write_spins_constraints(self, file_name="spins.in", cwd=None, spins_str=None
             s.logger.debug("No magnetic moments")
 
 
+class Group(dict):
+    def __init__(self, *args, **kw):
+        super(Group, self).__init__(*args, **kw)
+
+    def __setitem__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    def __setattr__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    __getattr__ = dict.get
+    # __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def set(self, name, content):
+        self[name] = content
+
+    def set_group(self, name, content=None):
+        if content is None:
+            content = Group()
+        self.set(name, content)
+
+    def set_flag(self, flag, val=True):
+        self.set(flag, val)
+
+    def set_parameter(self, parameter, val):
+        self.set(parameter, val)
+
+    def remove(self, name):
+        del self[name]
+
+    def remove_group(self, group):
+        self.remove(group)
+
+    def remove_flag(self, flag):
+        self.remove(flag)
+
+    def remove_parameter(self, parameter):
+        self.remove(parameter)
```

Comment:
why do you provide so many remove functions when they all do the same? Wouldn't 
```remove(whatever)```
not be much simpler?

Reply:
Yes, `remove(whatever)` was also still possible. I just wanted symmetry between all the `set()` and `remove()` functions, probably because I am just OCD :)

But as of 38bbfaaa I got rid of everything but the `remove()` function.